common/src/core/java/xyz/wagyourtail/jsmacros/core/library/impl/FReflection.java

```diff
@@ -489,6 +490,33 @@ public <T> WrappedClassInstance<T> wrapInstace(T instance) {
     public WrappedClassInstance<?> getWrappedClass(String className) throws ClassNotFoundException {
         return new WrappedClassInstance(null, Class.forName(className.replace("/", ".")));
     }
+
+    /**
+     * @param obj           the object to get the fields of
+     * @param includeStatic whether to include static fields
+     * @return a map of all fields and their respective values in the given object.
+     *
+     * @since 1.8.4
+     */
+    public Map<String, Object> toFieldMap(Object obj, boolean includeStatic) {
+        Map<String, Object> fields = new HashMap<>();
+        Class<?> clazz = obj.getClass();
+        while (clazz != null) {
+            for (Field field : clazz.getDeclaredFields()) {
+                if (Modifier.isStatic(field.getModifiers()) && !includeStatic) {
+                    continue;
+                }
+                field.setAccessible(true);
+                try {
+                    fields.put(field.getName(), field.get(obj));
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+            }
+            clazz = clazz.getSuperclass();
+        }
+        return fields;
+    }
```

Comment:
maybe `toFieldMap` should be split so just another one with a simple `getFields` instead of the recursive `getDeclaredFields`

thus the functions `toFieldMap` and `toDeclaredFieldMap`

Reply:
I deleted the method for now. The Reflection library in general should be overhauled to set fields, methods and constructors accessible, set and get field values directly, etc.
This method should then be added together with these other suggestions.